scraping/scraping/spiders/zalando.py

```diff
@@ -35,14 +36,15 @@ def parse_SERP(
         # Save HTML to database
         self._save_SERP(response)
 
-        # ads are less often than real products
-        article_elements = response.css("article::attr(class)").getall()
-        most_class_selector = max(set(article_elements), key=article_elements.count)
-        all_product_links = response.css(f"[class='{most_class_selector}']>a::attr(href)").getall()
-        all_product_links = list(set(all_product_links))
+        # Extract links from json object
+        data = json.loads(response.css("script.re-data-el-hydrate::text").get())
+        all_product_links = []
+        for key, value in data.get("graphqlCache", {}).items():
+            url = value.get("data", {}).get("product", {}).get("uri")
+            if url is not None:
+                all_product_links.append(url)
```

Comment:
An alternative for extracting the product links would be this oneliner:
`all_product_links = list(set(re.findall(r'https://www\.zalando\.\w{2}/[\w\d-]+\.html', response.text)))`
It basically relies on products having an ending of `.html` and no slashes in the url.

Reply:
Would it be easier to ignore whether or not a product is an ad?

This looks to me as very error prone regarding future changes.

Reply:
If you mean by ads products of the category `outfits` these are not part of both approaches. The one I commited also makes use of a JSON which comes from the API, so it includes only the products requested and with the regex approach these are not part of the results set, because they include a slash in the URL e.g. `https://www.zalando.de/outfits/CwOBJ4SiSu2/`

Reply:
And they also do not end on `.html`